drivers/mpsl/clock_control/clock_control_mpsl.c

```diff
@@ -331,6 +367,47 @@ static int clock_async_start(struct device *dev, clock_control_subsys_t subsys,
 	return errcode;
 }
 
+static void started_cb(struct device *dev,
+		       clock_control_subsys_t subsys,
+		       void *user_data)
+{
+	onoff_notify_fn notify = user_data;
+
+	enum clock_control_nrf_type type = (enum clock_control_nrf_type) subsys;
+	struct onoff_manager *mgr = &onoff_managers[type];
+
+	/* Result will always be zero since clock start cannot fail. */
+	const int res = 0;
+	notify(mgr, res);
+}
+
+static void onoff_start(struct onoff_manager *mgr, onoff_notify_fn notify)
```

Comment:
Per discussion offline, we could take the advantage of onoff service to maintain reference counting and client list of clock driver. To support the legacy applications (which haven't used on-off service) in this transaction period, we can keep the driver API (e.g. clock_async_start) as wrappers controlling onoff service.

There were some discussions over #1928.

Reply:
Thanks. Yes, and as discussed, we will do this in a later stage to avoid the larger diff 👍